src/DNSC/DNSUtil.hs

```diff
@@ -85,3 +87,24 @@ isRecvSendMsg = True
 #else
 isRecvSendMsg = False
 #endif
+
+---
+
+lookupRaw :: TimeStamp -> DNS.Resolver -> DNS.Domain -> DNS.TYPE -> IO (Either DNS.DNSError DNSMessage)
+isExtendedLookup :: Bool
+#if EXTENDED_LOOKUP
+lookupRaw now rslv dom typ = DNS.lookupRawRecv rslv dom typ mempty rcv
+  where
+    rcv sock = do
+      bs <- Socket.recv sock bufsiz
+      case DNS.decodeAt now bs of
+        Left  e   -> E.throwIO e
+        Right msg -> return msg
+    bufsiz = 16384 -- maxUdpSize in dns package, internal/Network/DNS/Types/Internal.hs
+
+isExtendedLookup = True
+#else
+lookupRaw _ = DNS.lookupRaw
+
+isExtendedLookup = False
+#endif
```

Comment:
extended interface, lookupRawRecv uses injected timestamp.
not extended, DNS.lookupRaw get current timestamp internally.